src/libaktualizr/primary/aktualizr.cc

```diff
@@ -67,13 +67,18 @@ void Aktualizr::UptaneCycle() {
   }
 
   uptane_client_->uptaneInstall(download_result.updates);
-  uptane_client_->putManifest();
+
   if (uptane_client_->isInstallCompletionRequired()) {
     // If there are some pending updates then effectively either reboot (ostree) or aktualizr restart (fake pack mngr)
     // is required to apply the update(s)
     LOG_INFO << "About to exit aktualizr so the pending updates can be applied after reboot";
     Shutdown();
   }
+
+  if (!uptane_client_->hasPendingUpdates()) {
+    // Why do we need to call putManifest here if it's called implicitly/indirectly at CheckUpdates().get() ?
```

Comment:
This is something of a legacy concept. We used to make a more explicit distinction between sending the manifest and checking for updates. Maybe we'll want to re-emphasize that distinction again some day. However, it is still relevant here when running in "automatic" mode with `UptaneCycle`, particularly if `config_.uptane.polling_sec` is set to a large value. Sending the manifest is necessary to tell the server that we've completed an update, and if it's going to be many minutes or hours before we next check for updates, we need to explicitly send the manifest here. Of course, with the recent reboot logic changes, maybe we should reconsider how we do that.

Reply:
If there is no any update the control flow never reaches this code.
If there is update(s) but it's pending and waiting for a reboot we don't wont to send any manifest anyway.
Just after reboot the manifest is sent just after update is completed from finalizeAfterReboot() at the initialization state.
The only case when it makes sense is when an update installation doesn't require any reboot (no pending updates) .

Reply:
That matches my understanding. Note that this may be more common than expected because of updates to secondaries that will not require rebooting the primary.

Reply:
ok